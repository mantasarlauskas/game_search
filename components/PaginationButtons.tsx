import { RefObject } from 'react';
import PaginatorButton from 'components/PaginatorButton';
import BackToTopButton from 'components/BackToTopButton';

function PaginationButtons({
    hasNextPage,
    hasPreviousPage,
    isFetching,
    paginatorRef,
    onClick,
}: PaginationButtonsProps) {
    return (
        <>
            <PaginatorButton
                ref={paginatorRef}
                isFetching={isFetching}
                isVisible={hasNextPage}
                onClick={onClick}
            />
            <BackToTopButton
                isEnabled={hasPreviousPage}
            />
        </>
    );
}

interface PaginationButtonsProps {
    hasNextPage: boolean;
    hasPreviousPage: boolean;
    isFetching: boolean;
    onClick?: () => void;
    paginatorRef?: RefObject<HTMLButtonElement>;
}

export default PaginationButtons;
