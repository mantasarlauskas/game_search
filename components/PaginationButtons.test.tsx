import { render, screen } from '@testing-library/react';
import PaginationButtons from 'components/PaginationButtons';
import { createRef } from 'react';

describe('<PaginationButtons />', () => {
    const props = {
        hasNextPage: true,
        hasPreviousPage: true,
        isFetching: false,
        paginatorRef: createRef<HTMLButtonElement>(),
        onClick: jest.fn(),
    };

    it('matches snapshot', () => {
        const { container } = render(<PaginationButtons {...props} />);
        expect(container).toMatchSnapshot();
    });

    it('shows both buttons', () => {
        render(<PaginationButtons {...props} />);
        expect(screen.getByRole('button', { name: 'Load more' })).toBeInTheDocument();
        expect(screen.getByRole('button', { name: 'Back to top' })).toBeInTheDocument();
    });
});
