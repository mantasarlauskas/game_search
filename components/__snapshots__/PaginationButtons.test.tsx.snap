// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PaginationButtons /> matches snapshot 1`] = `
<div>
  <button
    class="root"
    type="button"
  >
    Load more
  </button>
  <div
    class="root hidden"
    role="button"
    tabindex="0"
  >
    Back to top
  </div>
</div>
`;
